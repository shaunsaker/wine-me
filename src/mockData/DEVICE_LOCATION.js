const DEVICE_LOCATION = {
  latitude: -34.9263749,
  longitude: 18.9319848,
};

export default DEVICE_LOCATION;
