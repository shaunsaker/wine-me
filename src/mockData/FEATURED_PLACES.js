import PLACE from './PLACE';

const placeID = PLACE.id;
const FEATURED_PLACES = {
  1: {
    place_id: placeID,
  },
};

export default FEATURED_PLACES;
