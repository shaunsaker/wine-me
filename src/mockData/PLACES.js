import PLACE from './PLACE';

const PLACES = { 1: PLACE };

export default PLACES;
