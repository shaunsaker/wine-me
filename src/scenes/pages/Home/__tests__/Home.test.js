import React from 'react';
import renderer from 'react-test-renderer';

import { Home } from '..';
import FEATURED_PLACES from '../../../../mockData/FEATURED_PLACES';
import PLACES from '../../../../mockData/PLACES';
import DEVICE_LOCATION from '../../../../mockData/DEVICE_LOCATION';

jest.mock('../../../../components/PlaceList', () => 'PlaceList');
jest.mock('../../../../components/TabBar', () => 'TabBar');

describe('Home', () => {
  const spies = [];
  const dispatch = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(
        <Home featuredPlaces={FEATURED_PLACES} places={PLACES} deviceLocation={DEVICE_LOCATION} />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the loading state', () => {
      const component = renderer.create(
        <Home featuredPlaces={{}} places={{}} deviceLocation={DEVICE_LOCATION} />,
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle onSearchButtonPress', () => {
      spies[0] = jest.spyOn(Home.prototype, 'navigate');
      const component = renderer.create(
        <Home featuredPlaces={FEATURED_PLACES} places={PLACES} deviceLocation={DEVICE_LOCATION} />,
      );
      const instance = component.getInstance();

      instance.onSearchButtonPress();

      expect(spies[0]).toHaveBeenCalledWith('search');
    });

    it('should handle navigate', () => {
      const component = renderer.create(
        <Home featuredPlaces={FEATURED_PLACES} places={PLACES} deviceLocation={DEVICE_LOCATION} />,
      );
      const instance = component.getInstance();

      instance.navigate();
    });
  });

  describe('actions', () => {
    it('should call onSearchButtonPress on searchInput button press', () => {
      spies[0] = jest.spyOn(Home.prototype, 'onSearchButtonPress');
      const component = renderer.create(
        <Home featuredPlaces={FEATURED_PLACES} places={PLACES} deviceLocation={DEVICE_LOCATION} />,
      );
      const { root } = component;
      const targetComponent = root.findByProps({ testID: 'home.button.searchInput' });

      targetComponent.props.handlePress();

      expect(spies[0]).toHaveBeenCalled();
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
