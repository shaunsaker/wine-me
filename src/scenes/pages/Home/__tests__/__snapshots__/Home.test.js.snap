// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders renders the loading state 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingBottom": 0,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#48466D",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#48466D",
          "paddingBottom": 10,
          "paddingHorizontal": 15,
          "paddingTop": 10,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0, 0, 0, 0.70)",
          "borderRadius": 5,
          "flex": 1,
          "flexDirection": "row",
          "height": 48,
          "opacity": 1,
          "paddingHorizontal": 15,
        }
      }
      testID="home.button.searchInput"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
              "marginRight": 7.5,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Nunito-Regular",
            "fontSize": 17,
          }
        }
      >
        Search
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingLeft": 15,
          "paddingVertical": 10,
        }
      }
      testID="home.button.info"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={Object {}}
    style={Object {}}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginHorizontal": 15,
            "marginVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#212121",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 17,
            }
          }
        >
          Featured
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "height": 200,
            "justifyContent": "center",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#3D84A8"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginHorizontal": 15,
            "marginVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#212121",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 17,
            }
          }
        >
          Closest to me
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "height": 200,
            "justifyContent": "center",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#3D84A8"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>
  <TabBar />
</View>
`;

exports[`Home renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingBottom": 0,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#48466D",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#48466D",
          "paddingBottom": 10,
          "paddingHorizontal": 15,
          "paddingTop": 10,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0, 0, 0, 0.70)",
          "borderRadius": 5,
          "flex": 1,
          "flexDirection": "row",
          "height": 48,
          "opacity": 1,
          "paddingHorizontal": 15,
        }
      }
      testID="home.button.searchInput"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
              "marginRight": 7.5,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Nunito-Regular",
            "fontSize": 17,
          }
        }
      >
        Search
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingLeft": 15,
          "paddingVertical": 10,
        }
      }
      testID="home.button.info"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={Object {}}
    style={Object {}}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginHorizontal": 15,
            "marginVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#212121",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 17,
            }
          }
        >
          Featured
        </Text>
      </View>
      <PlaceList
        data={
          Array [
            Object {
              "address": "Stellenbosch, South Africa",
              "id": "1",
              "location": Object {
                "lat": -33.9263749,
                "lng": 18.9319848,
              },
              "name": "Bartinney Tasting Shed",
              "openingHours": Array [
                "Monday: 12:00 – 5:00 PM",
                "Tuesday: 12:00 – 5:00 PM",
                "Wednesday: 12:00 – 5:00 PM",
                "Thursday: 12:00 – 5:00 PM",
                "Friday: 12:00 – 8:00 PM",
                "Saturday: 10:00 AM – 12:30 PM",
                "Sunday: Closed",
              ],
              "phoneNumber": "021 885 1013",
              "photoReferences": Array [
                "CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA",
              ],
              "website": "http://www.bartinney.co.za",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginHorizontal": 15,
            "marginVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#212121",
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 17,
            }
          }
        >
          Closest to me
        </Text>
      </View>
      <PlaceList
        data={
          Array [
            Object {
              "address": "Stellenbosch, South Africa",
              "id": "1",
              "location": Object {
                "lat": -33.9263749,
                "lng": 18.9319848,
              },
              "name": "Bartinney Tasting Shed",
              "openingHours": Array [
                "Monday: 12:00 – 5:00 PM",
                "Tuesday: 12:00 – 5:00 PM",
                "Wednesday: 12:00 – 5:00 PM",
                "Thursday: 12:00 – 5:00 PM",
                "Friday: 12:00 – 8:00 PM",
                "Saturday: 10:00 AM – 12:30 PM",
                "Sunday: Closed",
              ],
              "phoneNumber": "021 885 1013",
              "photoReferences": Array [
                "CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA",
              ],
              "relativeDistance": 111.19492664455873,
              "website": "http://www.bartinney.co.za",
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
  <TabBar />
</View>
`;
